test/standalone/Vegetation/test_pmodel.jl

```diff
@@ -216,3 +223,251 @@ end
+    Weibull_param = FT(4) # unitless
+    a = FT(0.05 * 0.0098) # 1/m
+    plant_ν = FT(0.1) # m3/m3
+    plant_S_s = FT(1e-2)
+    conductivity_model =
+        PlantHydraulics.Weibull{FT}(K_sat_plant, ψ63, Weibull_param)
+    retention_model = PlantHydraulics.LinearRetentionCurve{FT}(a)
+    compartment_midpoints = [h_canopy]
+    compartment_surfaces = [FT(0.0), h_canopy]
+    # set rooting_depth param to largest possible value to test no roots
+    param_set = PlantHydraulics.PlantHydraulicsParameters(;
+        ai_parameterization = ai_parameterization,
+        ν = plant_ν,
+        S_s = plant_S_s,
+        rooting_depth = maxintfloat(FT),
+        conductivity_model = conductivity_model,
+        retention_model = retention_model,
+    )
+
+    transpiration = DiagnosticTranspiration{FT}()
+    soil_driver = PrescribedGroundConditions{FT}()
+    plant_hydraulics = PlantHydraulics.PlantHydraulicsModel{FT}(;
+        parameters = param_set,
+        transpiration = transpiration,
+        n_stem = n_stem,
+        n_leaf = n_leaf,
+        compartment_surfaces = compartment_surfaces,
+        compartment_midpoints = compartment_midpoints,
+    )
+
+    earth_param_set = LP.LandParameters(FT)
+    z_0m = FT(2.0) # m, Roughness length for momentum
+    z_0b = FT(0.1) # m, Roughness length for scalars
+    h_int = FT(30.0) # m, "where measurements would be taken at a typical flux tower of a 20m canopy"
+    shared_params = SharedCanopyParameters{FT, typeof(earth_param_set)}(
+        z_0m,
+        z_0b,
+        earth_param_set,
+    )
+    start_date = DateTime(2005)
+
+    # Dummy radiative forcing
+    zenith_angle =
+        (t, s) -> default_zenith_angle(
+            t,
+            s;
+            insol_params = earth_param_set.insol_params,
+            latitude = lat,
+            longitude = long,
+        )
+    shortwave_radiation = (t; latitude=lat, longitude=long, insol_params=earth_param_set.insol_params) -> 1000
+    longwave_radiation = t -> 200
+    radiation = PrescribedRadiativeFluxes(
+        FT,
+        TimeVaryingInput(shortwave_radiation),
+        TimeVaryingInput(longwave_radiation),
+        start_date;
+        θs = zenith_angle,
+        earth_param_set = earth_param_set,
+    )
+
+    # Dummy atmospheric forcing 
+    u_atmos = t -> 10 #m.s-1
+    liquid_precip = (t) -> 0 # m
+    snow_precip = (t) -> 0 # m
+    T_atmos = t -> 290 # Kelvin
+    q_atmos = t -> 0.001 # kg/kg
+    P_atmos = t -> 1e5 # Pa
+    h_atmos = h_int # m
+    c_atmos = (t) -> 4.11e-4 # mol/mol
+    atmos = PrescribedAtmosphere(
+        TimeVaryingInput(liquid_precip),
+        TimeVaryingInput(snow_precip),
+        TimeVaryingInput(T_atmos),
+        TimeVaryingInput(u_atmos),
+        TimeVaryingInput(q_atmos),
+        TimeVaryingInput(P_atmos),
+        start_date,
+        h_atmos,
+        earth_param_set;
+        c_co2 = TimeVaryingInput(c_atmos),
+    )
+
+    canopy = ClimaLand.Canopy.CanopyModel{FT}(;
```

Comment:
You can use the constructor Kat and I recently added to make this test much shorter. E.g. see the model setup in [this test](https://github.com/CliMA/ClimaLand.jl/blob/b3fce30e5e9b4e3024f0bbfd13e2e9654b541296/test/standalone/Vegetation/canopy_model.jl#L1361-L1410)

Reply:
Hmm it should be working with lat/lon domains. If you're still running into problems with it after rebasing on current main can you show me the error?